File: leaf similar trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum=0;
    int abs(int a){
        if(a<0){return -a;}
        return a;
    }
    int dfs(TreeNode* root){
        if(root==nullptr){return 0;}
        int l=dfs(root->left);
        int r=dfs(root->right);
        sum+=abs(l-r);
        return root->val+l+r;
    }
    int findTilt(TreeNode* root) {
        if(root==nullptr){return sum;}
        dfs(root);
        return sum;
    }
};

File: leaf similar trees.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
   void dfs(TreeNode root,ArrayList<Integer>arr){
       if(root==null){return ;}
       if(root.left==null&&root.right==null){arr.add(root.val);return;}
       dfs(root.left,arr);
       dfs(root.right,arr);
       return;
   }
    public boolean leafSimilar(TreeNode root1, TreeNode root2) {
        ArrayList<Integer>v1=new ArrayList<Integer>();
        ArrayList<Integer>v2=new ArrayList<Integer>();
        dfs(root1,v1);
        dfs(root2,v2);
        if(v1.size()!=v2.size()){return false;}
        for(int i=0;i<v1.size();i++){
            if(v1.get(i)!=v2.get(i)){return false;}
        }
        return true;
    }
}